OMEdit/OMEditLIB/Simulation/SimulationDialog.cpp

```diff
@@ -1706,6 +1706,20 @@ void SimulationDialog::performSimulation(const SimulationOptions &simulationOpti
   QApplication::restoreOverrideCursor();
   mIsReSimulate = false;
   if (isTranslationSuccessful) {
+    // run post compilation command after a successful translateModel call
+    const QString postCompilationCommand = OptionsDialog::instance()->getSimulationPage()->getPostCompilationCommand();
+    if (postCompilationCommand.size())
+    {
+      QProcess process;
+      process.setWorkingDirectory(OptionsDialog::instance()->getGeneralSettingsPage()->getWorkingDirectory());
+      process.start(postCompilationCommand);
+      const bool ok = process.waitForFinished(-1);
+      if (!ok)
+      {
+        QString msg = tr("Post compilation command failed <b>%1</b>.").arg(postCompilationCommand);
+        MessagesWidget::instance()->addGUIMessage(MessageItem(MessageItem::Modelica, msg, Helper::scriptingKind, Helper::notificationLevel));
+      }
+    }
```

Comment:
This is not the right place. Until here the model is only translated and no executable is built that you can call post compilation command on.

You need to move this code in `SimulationOutputWidget`. See for example `compileModel()` method.

Do not call `QProcess` in a blocking state because the post compilation command can hang forever. Using a better time out is just a work around and doesn't guarantee the correct result.
See how the compilation process is called. We run the compilation process and connect to its finished signal. If the process is finished successfully then we call simulation process otherwise user can see the error or can cancel compilation. So you need do something similar. Add a new process routine in between the compilation and simulation processes.

Reply:
thanks for the feedback, I'll try to get the error handling working properly

Reply:
This is great advice, thanks!
I can now interrupt if the process hangs and get any error message, and the status is printed into the compilation log.